In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import ttest_ind, shapiro, levene, f_oneway
from statsmodels.stats.multicomp import pairwise_tukeyhsd

In [7]:
#Load the data
df = pd.read_csv("https://raw.githubusercontent.com/vincentarelbundock/Rdatasets/master/csv/datasets/PlantGrowth.csv")

df.to_csv("PlantGrowth.csv", index=False)

print(df.head())

print(df.describe())

print(df["group"].value_counts())

   rownames  weight group
0         1    4.17  ctrl
1         2    5.58  ctrl
2         3    5.18  ctrl
3         4    6.11  ctrl
4         5    4.50  ctrl
        rownames     weight
count  30.000000  30.000000
mean   15.500000   5.073000
std     8.803408   0.701192
min     1.000000   3.590000
25%     8.250000   4.550000
50%    15.500000   5.155000
75%    22.750000   5.530000
max    30.000000   6.310000
group
ctrl    10
trt1    10
trt2    10
Name: count, dtype: int64


This dataset has 30 observations 10 in each group (ctrl, trt1, trt2).
The plantweight mean is 5.07 and has a standard deviation of 0.70.

A t-test is a statistical test used to determine if there is a significant difference between the means of 2 groups.
It can be used to compare the means of two small samples.

The Null Hypothesis is: There is no difference between the means of the two groups.
The Alternate Hypothesis: There is a difference between the means
The test statistic is calculated using the formula.
The result is compared against the critical value based on the degree of freedom and significance level.
The p-value is then used to interpret result. A value less than the chosen cut-off (e.g 0.05) indicates statistical significance. 


The type of t-test we will be carrying out is an Independent group t-test. This is because the trt1 and trt2 groups consist of separate sets of plants that are not related to each other.
There are also no paired observations.

A t-test works with the following assumptions:
Normality:
The data in each group are approximately normally distributed.

Independent Samples:
Observations in one group are not related to those in the other.

Homogeneity of Variances:
The variance of the two groups are similar.

Scale of Measurement:
The dependent variable are measured on a continuous scale.